test/distributed/fsdp/test_fsdp_tp_integration.py

```diff
@@ -373,6 +373,14 @@ def forward(self, x):
         self.assertEqual(comm_counts[funcol.all_gather_into_tensor], 2)
         self.assertEqual(comm_counts[funcol.all_reduce], 1)
 
+        grads = [p.grad for p in fsdp_2d_model.parameters() if p.grad is not None]
+
+        for grad in grads:
+            if isinstance(grad, DTensor):
+                grad = grad.full_tensor()
```

Comment:
In the current 2D design, do we see a case when a gradient is DTensor? They are all expected to be flattened tensors IIUC. Curious to learn how this is possible?

Reply:
Oh yes that's right! I forgot to update the test case, will update soon